client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -65,18 +65,28 @@ tie.factory('PythonPrereqCheckService', [
     var detectAndGetWrongLanguageType = function(code) {
       var codeLines = getNonStringLines(code);
 
+      var rawCodeArray = code.split('\n');
       for (var i = 0; i < WRONG_LANGUAGE_ERRORS.python.length; i++) {
         var error = WRONG_LANGUAGE_ERRORS.python[i];
         var regexp = new RegExp(error.regExString);
 
         if (error.allowMultiline) {
           if (code.search(regexp) !== -1) {
-            return error.errorName;
+            var errorFirstLine = code.search(regexp);
+            for (var j = 0; errorFirstLine > 0; j++) {
+              var sub = rawCodeArray[j].length;
+              errorFirstLine -= sub;
+            }
+            error.errorLine = errorFirstLine;
+            return error;
           }
         } else {
-          for (var j = 0; j < codeLines.length; j++) {
-            if (codeLines[j].search(regexp) !== -1) {
-              return error.errorName;
+          for (var k = 0; k < codeLines.length; k++) {
+            if (codeLines[k].search(regexp) !== -1) {
+              error.errorLine = rawCodeArray.findIndex(function(el) {
+                return el === codeLines[k];
```

Comment:
I think, in order to do this properly, we should introduce a mapping from rawCodeArray to codeLines? Multiple lines may have the same code.

Reply:
Error returns at the first instance found. At this point, we are only concerned with one error at a time.
We can revisit if we want to show all errors at once.
Done.